frontend/app/src/store/blockchain/balances/chains.ts

```diff
@@ -39,11 +39,21 @@ export const useChainBalancesStore = defineStore('balances/chain', () => {
       return;
     }
 
+    const chainValues = perAccount[chain] ?? {};
+
     set(balances, {
       ...get(balances),
-      [chain]: perAccount[chain] ?? {}
+      [chain]: chainValues
     });
 
+    // todo: this is temporary, to update the tokens count
+    // todo: remove when BE updates the endpoint to refresh detected tokens
+    if (chain === Blockchain.OPTIMISM) {
+      const { updateStateOnBalanceRefresh } = useBlockchainTokensStore();
```

Comment:
Can we move this outside the update function scope and into the store setup scope?

Initializing stores/composables outside of the setup can be problematic.

Reply:
Okay, will do

Reply:
done.